consensus/src/experimental/commit_phase.rs

```diff
@@ -2,86 +2,89 @@
-  │            └──────►│              │          │   │                      │
-  │         Enqueue    └──────────┬───┘          │   ▼                      │
-  │                               │              │ Check if committable:    │
-  └─────────────── Check Channels │              │    If so, commit and dequeue
-                                  │              │
-                                  └────────► Main Loop
-                                                 ▲
-                                                 │
-                                             fn start
-
- Commit phase takes in the executed blocks from the execution
- phase and commit them. Specifically, commit phase signs a commit
- vote message containing the execution result and broadcast it.
- Upon collecting a quorum of agreeing votes to a execution result,
- the commit phase commits the blocks as well as broadcasts a commit
- decision message together with the quorum of signatures. The commit
- decision message helps the slower nodes to quickly catch up without
- having to collect the signatures.
- */
-
-#[derive(Clone)]
-struct PendingBlocks {
+Commit phase takes in the executed blocks from the execution
+phase and commit them. Specifically, commit phase signs a commit
+vote message containing the execution result and broadcast it.
+Upon collecting a quorum of agreeing votes to a execution result,
+the commit phase commits the blocks as well as broadcasts a commit
+decision message together with the quorum of signatures. The commit
+decision message helps the slower nodes to quickly catch up without
+having to collect the signatures.
+*/
+
+pub struct PendingBlocks {
     vecblocks: Vec<ExecutedBlock>,
     ledger_info_sig: LedgerInfoWithSignatures,
+    block_info: BlockInfo,
 }
 
 impl PendingBlocks {
     pub fn new(vecblocks: Vec<ExecutedBlock>, ledger_info_sig: LedgerInfoWithSignatures) -> Self {
+        assert!(!vecblocks.is_empty()); // the commit phase should not accept empty blocks.
+        let block_info = vecblocks.last().unwrap().block_info();
         Self {
             vecblocks,
             ledger_info_sig,
+            block_info,
         }
     }
+
+    pub fn block_info(&self) -> &BlockInfo {
+        &self.block_info
+    }
+
+    pub fn round(&self) -> u64 {
+        self.block_info().round()
+    }
+
+    pub fn vecblocks(&self) -> &Vec<ExecutedBlock> {
+        &self.vecblocks
+    }
+
+    pub fn ledger_info_sig(&self) -> &LedgerInfoWithSignatures {
+        &self.ledger_info_sig
+    }
+
+    pub fn ledger_info_sig_mut(&mut self) -> &mut LedgerInfoWithSignatures {
+        &mut self.ledger_info_sig
+    }
+
+    pub fn replace_ledger_info_sig(&mut self, new_ledger_info_sig: LedgerInfoWithSignatures) {
+        self.ledger_info_sig = new_ledger_info_sig
+    }
+
+    pub fn verify(&self, verifier: &ValidatorVerifier) -> ::std::result::Result<(), VerifyError> {
+        self.ledger_info_sig.verify_signatures(verifier)
```

Comment:
we need to verify the ledger info's block info matches the last block

Reply:
resolved